src/agent/initdata-processor/src/initdata.rs

```diff
@@ -0,0 +1,117 @@
+use std::{
+    os::unix::fs::FileTypeExt,
+    path::{Path, PathBuf},
+};
+
+use anyhow::Result;
+use async_compression::tokio::bufread::GzipDecoder;
+use slog::{error, info, Logger};
+use tokio::io::{AsyncReadExt, AsyncSeekExt};
+use tokio::task::JoinHandle;
+
+/// Magic number of initdata device
+pub const INITDATA_MAGIC_NUMBER: &[u8] = b"initdata";
+
+/// It's designed to be run in a separate tokio task to check if a the potential device is the initdata device.
+async fn check_initdata_device(logger: Logger, path: PathBuf) -> Result<Option<String>> {
+    let metadata = match tokio::fs::metadata(&path).await {
+        Ok(m) => m,
+        Err(_) => return Ok(None),
+    };
```

Comment:
for this you can do the let else pattern, which makes things a bit more legible:

```rust
let Ok(metadata) = tokio::fs::metadata(&path).await else {
   return Ok(None);
}
```